### PR TITLE
[13.0][IMP]report_xlsx: posibility to use boilerplate template on xlsx report

### PR DESCRIPTION
**Pull request by GuillemCForgeFlow**
These changes add the possibility to define a boilerplate template from which to start working from. From there, you can add the needed extra information for the inherited report.

cc @ForgeFlow

### CODE REVIEW

**Comment by GuillemCForgeFlow** [comment]
@adrienpeiffer @sebalix @sbidoul @etobella 
Perhaps I could get a review here? 😃

**Review by JordiMForgeFlow — APPROVED** [review]
Code review + functionally tested 👍🏼

**Review by HviorForgeFlow — APPROVED** [review]
Code review :+1:

File: report_xlsx/report/report_abstract_xlsx.py [hunk 1]
```diff
@@ -5,7 +5,14 @@
 import re
 from io import BytesIO
 
-from odoo import models
+import openpyxl
```
Comment (etobella) [inline report_xlsx/report/report_abstract_xlsx.py:8]: you should add a try except just to not break old installations

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
This change is not valid for a stable version. It's changing the dependent library. It should be added on an extra module, not here.

**Comment by GuillemCForgeFlow** [comment]
> This change is not valid for a stable version. It's changing the dependent library. It should be added on an extra module, not here.

Ok, then I will add the feature in a new module.

**Comment by sbidoul** [comment]
> This change is not valid for a stable version. It's changing the dependent library. It should be added on an extra module, not here.

I don't think we have such policy in OCA. With a major version bump we can change dependencies.

**Comment by pedrobaeza** [comment]
No, Stéphane, if this is merged, it's breaking existing installations, so that's not acceptable. One thing is to change datamodel providing migration scripts, and another to totally break the module to existing users.

**Comment by sbidoul** [comment]
Then we disagree on this. Such dependency refactorings did happen in the past and should continue be allowed IMO.

**Comment by pedrobaeza** [comment]
This module is the base for a lot of other modules (account_financial_report, account_asset_management, etc), and one thing is to add an extra feature that doesn't work if you don't have the dependency, and another to break the existing behavior on a version that lasts 4 years. I think there are degrees of acceptance, and this one is not acceptable.

**Comment by sbidoul** [comment]
Yes, I agree we must exercise common sense. I was talking about general policy, not specifics of this PR.

**Comment by pedrobaeza** [comment]
I haven't talked in any moment about a general OCA policy. I started saying "This change"...

**Comment by GuillemCForgeFlow** [comment]
closing this as the changes are not relevant anymore after the new module adding the functionality got created/merged: https://github.com/OCA/reporting-engine/pull/798

**Pull request closed without merging**